src/cache.ts

```diff
@@ -51,9 +51,60 @@ export interface CacheBackend {
 }
 
 export class FilesystemCacheBackend implements CacheBackend {
-    constructor(private cacheDir: string) {
-        fs.ensureDirSync(cacheDir);
+    constructor(private cacheDir: string, private sizeLimit: number | null) {
+        //create chacheDir if it doesn't exist
+        fs.mkdir(cacheDir, { recursive: true });
+
+        this.cleanup(); // don't await
+        setInterval(this.cleanup, 1000 * 60 * 15);
+    }
+
+    private async cleanup() {
```

Comment:
this is currently not safe for multiple replicas (out of scope for this PR, but we should create a TODO)